lib/imgix/rails/tag.rb

```diff
@@ -23,23 +23,25 @@ def self.parameters
     @@parameters
   end
 
-  def initialize(source, options={})
+  def initialize(source, tag_options: {}, url_params: {}, widths: [])
     @source = source
-    @options = options
+    @tag_options = tag_options
+    @url_params = url_params
+    @widths = widths.length > 0 ? widths : target_widths
   end
 
 protected
 
-  def srcset(opts=@options)
+  def srcset(url_params: @url_params, widths: @widths)
     @source = replace_hostname(@source)
-    widths = opts[:widths] || target_widths
+    widths = widths || target_widths
 
     widths.map do |width|
-      srcset_options = opts.slice(*self.class.available_parameters).except(:widths)
+      srcset_options = url_params.clone
```

Comment:
Can we rename this variable to `srcset_params` or `srcset_url_params` for clarity?